lib/apisession/session-store.ts

```diff
@@ -0,0 +1,83 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+import _ = require('lodash');
+import Session = require('./session');
+
+export = SessionStore;
+
+class SessionStore {
```

Comment:
This seems similar to `SubscriptionStore` that was introduced in #38.  To summarize our verbal discussion, let's provide a `Map` class that is is the same interfaces as the ES6 maps that can be extended.